api/payloads/service_binding_test.go

```diff
@@ -64,6 +78,32 @@ var _ = Describe("ServiceBindingCreate", func() {
 		apiError             errors.ApiError
 	)
 
+	When("binding is of type key", func() {
+		BeforeEach(func() {
+			serviceBindingCreate = new(payloads.ServiceBindingCreate)
+			createPayload = payloads.ServiceBindingCreate{
+				Relationships: &payloads.ServiceBindingRelationships{
+					ServiceInstance: &payloads.Relationship{
+						Data: &payloads.RelationshipData{
+							GUID: "service-instance-guid",
+						},
+					},
+				},
+				Type: "key",
+			}
+		})
+
+		JustBeforeEach(func() {
+			validatorErr = validator.DecodeAndValidateJSONPayload(createJSONRequest(createPayload), serviceBindingCreate)
+			apiError, _ = validatorErr.(errors.ApiError)
+		})
+
+		It("succeeds", func() {
```

Comment:
We probably need some negative test here - like when the required `name` field is omited